test/pylib/scylla_cluster.py

```diff
@@ -165,9 +169,31 @@ def make_scylla_conf(mode: str, workdir: pathlib.Path, host_addr: str, seed_addr
     '--logger-log-level', 'raft_topology=debug',
     '--logger-log-level', 'query_processor=debug',
     '--logger-log-level', 'group0_raft_sm=trace',
-    '--logger-log-level', 'group0_voter_handler=debug',
 ]
 
+# A path to a Scylla executable, with version-specific
+# (i.e. not supported by *all* relevant versions)
+# Scylla config attached.
+class ScyllaVersionDescription(NamedTuple):
+    path: str # path to the Scylla executable
+    config: dict # a dictionary of added scylla.yaml options
+    argv: list[str] # a list of added CLI args
+
+# Returns the description of the current version.
+# (I.e. the one we just built and are now testing).
+#
+# (The path has to be passed by an argument because the current executable
+# has no fixed location -- the build directory depends on the build mode,
+# and in the case of cmake can be moved to something different than `build`.)
+def get_current_version_description(path: str) -> ScyllaVersionDescription:
+    return ScyllaVersionDescription(
+        path=path,
+        config={},
+        argv=[
+            '--logger-log-level', 'group0_voter_handler=debug',
+        ]
```

Comment:
One could also handle it by passing the wanted log levels as a dictionary, querying the binary (with --help-loggers) about which loggers are supported, and dropping the unsupported ones. But having it vetted per version is also okay.

Reply:
It applies to more than loggers. It also applies to stuff like `--kernel-page-cache=1`.

I did think about what you are saying. (Filtering new options by `--help-*` rather than explicitly adding options per version). But while it would require less maintenance, it felt like the wrong kind of maintenance to automate.

E.g. imagine that an option is renamed/extended in a newer version. In this case, the "filtering" approach would stop setting the option in the older Scylla version, or set the option key to an unsupported option value. But maybe you don't want that, maybe you want to keep setting the option as before.